library/HTMLPurifier/AttrDef/HTML/Bool2.php

```diff
@@ -19,9 +19,10 @@ class HTMLPurifier_AttrDef_HTML_Bool2 extends HTMLPurifier_AttrDef_HTML_Bool
      */
     public function validate($string, $config, $context)
     {
+        $name = $this->name ? : $context->get('CurrentAttr');
```

Comment:
Please use full form of ternary operator, so it's [PHP 5.2 compatible](https://github.com/ezyang/htmlpurifier/blob/master/composer.json#L16), just as the original library.

I would use `strlen($this->name)` for the condition instead of just `$this->name`.